File: 160.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
        ListNode *cur1, *cur2;
        cur1=headA;
        cur2=headB;
        int cnt1=0;
        int cnt2=0;
        while(cur1!=NULL)
        {
            cnt1++;
            cur1=cur1->next;
        }
        while(cur2!=NULL)
        {
            cnt2++;
            cur2=cur2->next;
        }
        if(cnt1>=cnt2)
        {
            for(int i=0;i<cnt1-cnt2;i++) headA=headA->next;
        }
        else
        {
            for(int i=0;i<cnt2-cnt1;i++) headB=headB->next;
        }
        while(headA!=NULL)
        {
            if(headA==headB) return headA;
            headA=headA->next;
            headB=headB->next;
        }
        return NULL;
    }
};

File: 160.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def getIntersectionNode(self, headA: ListNode, headB: ListNode) -> Optional[ListNode]:
        cur1=headA
        cur2=headB
        cnt1=cnt2=0
        while cur1:
            cnt1+=1;
            cur1=cur1.next
        while cur2:
            cnt2+=1;
            cur2=cur2.next
        if cnt1>=cnt2:
            for i in range(cnt1-cnt2):
                headA=headA.next
        else:
            for i in range(cnt2-cnt1):
                headB=headB.next
        while headA:
            if headA==headB: return headA
            headA=headA.next
            headB=headB.next
        return None
